core/simulations/helpers/createVaultWithCollateral.ts

```diff
@@ -147,10 +147,52 @@ const giveJoinContractAllowance = async (collateralConfig: CollateralConfig, amo
     await contract.approve(addressJoin, amountRaw);
 };
 
-const createVaultWithCollateral = async (collateralType: CollateralType, collateralOwned: BigNumber) => {
+export const determineBalanceSlot = async (
+    collateralType: CollateralType
+): Promise<[string, 'vyper' | 'solidity'] | [null, null]> => {
+    console.info('Determining balance slot...');
+    const collateralConfig = getCollateralConfigByType(collateralType);
+    const tokenContractAddress = await getContractAddressByName(TEST_NETWORK, collateralConfig.symbol);
+    try {
+        const [slot, languageFormat] = await findERC20BalanceSlot(tokenContractAddress);
+        console.info(
+            `Balance slot is ${slot}, language format is ${languageFormat}, contract address is ${tokenContractAddress}`
+        );
+        return [slot, languageFormat];
+    } catch (e) {
+        if (
+            e instanceof Error &&
+            e.message.startsWith('Failed to find the slot of the balance for the token in address ')
+        ) {
+            return [null, null];
+        }
+        throw e;
+    }
+};
+
+const createVaultWithCollateral = async (
+    collateralType: CollateralType,
+    collateralOwned: BigNumber,
+    balanceSlot: string | null,
+    languageFormat: 'vyper' | 'solidity' | null
+) => {
     const collateralConfig = getCollateralConfigByType(collateralType);
-    await setAndCheckCollateralInVat(collateralType, collateralOwned);
-    await checkAndWithdrawCollateralFromVat(collateralConfig, collateralOwned);
+
+    const tokenContractAddress = await getContractAddressByName(TEST_NETWORK, collateralConfig.symbol);
+    if (balanceSlot && languageFormat) {
+        await setCollateralInWallet(
+            tokenContractAddress,
+            balanceSlot,
+            collateralOwned,
+            collateralConfig.decimals,
+            undefined,
+            languageFormat
+        );
```

Comment:
The api of the function is a nightmare (: 
Please create and use here a function `setCollateralInWallet(collateralType: string, collateralAmount: BigNumber, walletAddress?: string)`